crates/hir_def/src/body.rs

```diff
@@ -116,14 +119,14 @@ impl Expander {
         let call_id =
             macro_call.as_call_id_with_errors(db, self.def_map.krate(), resolver, &mut |e| {
                 err.get_or_insert(e);
-            });
+            })?;
         let call_id = match call_id {
             Some(it) => it,
             None => {
                 if err.is_none() {
```

Comment:
If `as_call_id_with_errors` return `Result<_>` , we could just make it to something like:
```rust
   let err = err.or_else(|| Some(ExpandError::UnresolveMacroCall));
```